src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/multi-cluster/multi-cluster-form/multi-cluster-form.component.ts

```diff
@@ -35,14 +45,61 @@ export class MultiClusterFormComponent implements OnInit, OnDestroy {
   ) {
     this.createForm();
   }
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    if (this.action === 'edit') {
+      this.remoteClusterForm.get('remoteClusterUrl').setValue(this.cluster.url);
+      this.remoteClusterForm.get('remoteClusterUrl').disable();
+      this.remoteClusterForm.get('clusterAlias').setValue(this.cluster.cluster_alias);
+    }
+    if (this.action === 'reconnect') {
+      this.remoteClusterForm.get('remoteClusterUrl').setValue(this.cluster.url);
+      this.remoteClusterForm.get('remoteClusterUrl').disable();
+      this.remoteClusterForm.get('clusterAlias').setValue(this.cluster.cluster_alias);
+      this.remoteClusterForm.get('clusterAlias').disable();
+      this.remoteClusterForm.get('username').setValue(this.cluster.user);
+      this.remoteClusterForm.get('username').disable();
+      this.remoteClusterForm.get('clusterFsid').setValue(this.cluster.name);
+      this.remoteClusterForm.get('clusterFsid').disable();
+    }
+    this.clusterAliasNames = this.clustersData.map((cluster) => {
+      return cluster['cluster_alias'];
+    });
+
+    this.clusterUrls = this.clustersData.map((cluster) => {
+      return cluster['url'];
+    });
+
+    this.clusterUsers = this.clustersData.map((cluster) => {
+      return cluster['user'];
+    });
```

Comment:
maybe something like this should work?
```suggestion
    [this.clusterAliasNames, this.clusterUrls, this.clusterUsers] = ['cluster_alias', 'url', 'user'].map(prop => this.clustersData?.map(cluster => cluster[prop]));
```